projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts

```diff
@@ -45,20 +44,16 @@ export class CartItemComponent implements OnInit, OnChanges {
     optionalBtn: null,
   };
 
+  @Input() promotions: Observable<PromotionResult[]>;
+
   appliedProductPromotions$: Observable<PromotionResult[]>;
   iconTypes = ICON_TYPE;
   readonly CartOutlets = CartOutlets;
 
-  constructor(
-    protected promotionService: PromotionService,
-    protected cartItemContextSource: CartItemContextSource
-  ) {}
+  constructor(protected cartItemContextSource: CartItemContextSource) {}
 
   ngOnInit() {
-    this.appliedProductPromotions$ = this.promotionService.getProductPromotionForEntry(
-      this.item,
-      this.promotionLocation
```

Comment:
it seems it called the `getProductPromotionForEntry` by passing promotionLocation as `ActiveCart`. Now, we just passed promotion, which may get from Order.